File: server/src/models/file.ts
import { Schema, model } from 'mongoose';

export interface IFile {
    fileName: string;
    hash: string;
    reputation: string;
    fileSize: number;// in byte
    createdAt: Date;
}

const fileSchema = new Schema<IFile>({
    fileName: { type: String, required: true, trim: true },
    hash: { type: String, required: true, unique: true },
    reputation: { type: String, required: true },
    fileSize: { type: Number, required: true, min: 0 },
    createdAt: { type: Date, default: Date.now, immutable: true }
});

const File = model<IFile>('File', fileSchema);

export default File;

File: go-server/models/file.go
package models

import (
	"time"
)

type File struct {
	FileName   string    `bson:"file_name" json:"file_name"`
	Hash       string    `bson:"hash" json:"hash"`
	Reputation int       `bson:"reputation" json:"reputation"`
	FileSize   int64     `bson:"file_size" json:"file_size"` // byte
	CreatedAt  time.Time `bson:"created_at" json:"created_at"`
}

// type File struct {
//     Key string `bson:"key" json:"key"`
//     Value string  `bson: "value" json :"value"`
// }
